src/lib/core/overlay/position/connected-position-strategy.ts

```diff
@@ -116,10 +117,11 @@ export class ConnectedPositionStrategy implements PositionStrategy {
       // (top, left) coordinate for the overlay at `pos`.
       let originPoint = this._getOriginConnectionPoint(originRect, pos);
       let overlayPoint = this._getOverlayPoint(originPoint, overlayRect, viewportRect, pos);
+      let overlayDimensions = this._getCSSDimensions(overlayRect, overlayPoint, pos);
```

Comment:
It's weird to have both `overlayPoint` and `overlayDimensions` since they're both kind of the same thing and it would be very easy to use the wrong one. 

What if we leave the calculations they way they are now, but then also pass the `ConnectionPositionPair` and the overlay `ClientRect` to `_setElementPosition` which figures out if it should set `bottom`/`right` there? I think it might be better, since then you're deferring the calculation until you have to know and don't have to create a type to pass it around.